Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -256,17 +255,16 @@ - (void) addCollection: (CBLCollection*)collection
         [NSException raise: NSInvalidArgumentException
                     format: @"Attempt to add an invalid collection"];
     }
-    
-    if (_collections.count > 0) {
-        if (_collections[0].db != collection.db) {
+    CBLDatabase* colDB = collection.db;
+    if (_database) {
```

Comment:
Technically, colDB could be nil at this point as well.

```
CBLDatabase* colDB = collection.db;
if (!colDB) {
    [NSException raise: NSInvalidArgumentException
                    format: @"Attempt to add an invalid collection"];
}
```

Reply:
```
CBLDatabase* colDB = collection.db;
    if (!collection.isValid || !colDB) {
        [NSException raise: NSInvalidArgumentException
                    format: @"Attempt to add an invalid collection"];
    }
    ```